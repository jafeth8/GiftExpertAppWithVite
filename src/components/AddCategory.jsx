import React, { useState } from 'react'


export const AddCategory = ({onNewValue}) => {

  const [inputValue, setInputValue] = useState("");

  const handleInputChange=(event)=>{
    setInputValue(event.target.value)
  }

  const handleSubmit=(e)=>{
    e.preventDefault();
    
    if(inputValue.trim().length<=1)return;
    onNewValue(inputValue.trim());
    setInputValue("");
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="buscar gif"
        value={inputValue}
        onChange={(e)=>handleInputChange(e)}
      />
    </form>
  )

}
