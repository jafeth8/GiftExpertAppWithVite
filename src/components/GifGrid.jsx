
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';


export const GifGrid = ({category}) => {
 
 const {dataGifs,isLoading} = useFetchGifs(category);
  
  return (
    <>
        <h3>{category}</h3>

        { isLoading && <h2>cargando...</h2> }

        <div className='card-grid'>
          {
            dataGifs.map((gif)=>(
              <GifItem key={gif.id} {...gif}/>
            ))
          }
        </div>
    </>
  )
}
