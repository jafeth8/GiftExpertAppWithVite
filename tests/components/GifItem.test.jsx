import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('test GifItem', () => {
    const title="shiro";
    const url="https://www.google.com.mx/shiro.jpg";

    test('debe hacer match con el snapshot', () => {
        const {container}=render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    })
    
    test('img debe mostrar url y alt correctamente', () => {
        render(<GifItem title={title} url={url}/>);
        const {src,alt}=screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        
    })

})