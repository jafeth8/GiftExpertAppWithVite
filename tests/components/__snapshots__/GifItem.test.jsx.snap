// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="shiro"
      src="https://www.google.com.mx/shiro.jpg"
    />
    <p>
      shiro
    </p>
  </div>
</div>
`;
